pkg/proxy/conntrack/conntrack_test.go

```diff
@@ -27,47 +27,63 @@ import (
 	v1 "k8s.io/api/core/v1"
 	"k8s.io/utils/exec"
 	fakeexec "k8s.io/utils/exec/testing"
-	utilnet "k8s.io/utils/net"
 )
 
-func familyParamStr(isIPv6 bool) string {
-	if isIPv6 {
-		return " -f ipv6"
-	}
-	return ""
+var success = func() ([]byte, []byte, error) { return []byte("1 flow entries have been deleted"), nil, nil }
+var nothingToDelete = func() ([]byte, []byte, error) {
+	return []byte(""), nil, fmt.Errorf("conntrack v1.4.2 (conntrack-tools): 0 flow entries have been deleted")
 }
 
-func TestExecConntrackTool(t *testing.T) {
-	fcmd := fakeexec.FakeCmd{
-		CombinedOutputScript: []fakeexec.FakeAction{
-			func() ([]byte, []byte, error) { return []byte("1 flow entries have been deleted"), nil, nil },
-			func() ([]byte, []byte, error) { return []byte("1 flow entries have been deleted"), nil, nil },
-			func() ([]byte, []byte, error) {
-				return []byte(""), nil, fmt.Errorf("conntrack v1.4.2 (conntrack-tools): 0 flow entries have been deleted")
-			},
-		},
+func makeCT(result fakeexec.FakeAction) (*fakeexec.FakeExec, *fakeexec.FakeCmd) {
+	fcmd := &fakeexec.FakeCmd{
+		CombinedOutputScript: []fakeexec.FakeAction{result},
 	}
 	fexec := &fakeexec.FakeExec{
 		CommandScript: []fakeexec.FakeCommandAction{
-			func(cmd string, args ...string) exec.Cmd { return fakeexec.InitFakeCmd(&fcmd, cmd, args...) },
-			func(cmd string, args ...string) exec.Cmd { return fakeexec.InitFakeCmd(&fcmd, cmd, args...) },
-			func(cmd string, args ...string) exec.Cmd { return fakeexec.InitFakeCmd(&fcmd, cmd, args...) },
+			func(cmd string, args ...string) exec.Cmd { return fakeexec.InitFakeCmd(fcmd, cmd, args...) },
 		},
 		LookPathFunc: func(cmd string) (string, error) { return cmd, nil },
 	}
 
-	testCases := [][]string{
-		{"-L", "-p", "udp"},
-		{"-D", "-p", "udp", "-d", "10.0.240.1"},
-		{"-D", "-p", "udp", "--orig-dst", "10.240.0.2", "--dst-nat", "10.0.10.2"},
-	}
+	return fexec, fcmd
+}
 
-	expectErr := []bool{false, false, true}
+func getExecutedCommand(fexec *fakeexec.FakeExec, fcmd *fakeexec.FakeCmd) string {
+	// FakeExec panics if you try to run more commands than you set it up for. So the
+	// only possibilities here are that we ran 1 command or we ran 0.
+	if fexec.CommandCalls != 1 {
+		return "(no command executed!)"
```

Comment:
I think is better passing the `t testing.T` as argument and fail here with `t.Errorf("unexpected command calls, expected 1 got %d", fexex.CommandCalls)` , instead of relying on our custom message

Reply:
but is a nit, if you are calling for the executed command is normal to assume you are going to parse the output

Reply:
you actually puts this inside the interface later

Reply:
> but is a nit, if you are calling for the executed command is normal to assume you are going to parse the output

in general, FakeExec is not very good at dealing with the case where the code being tested does not make exactly the set of calls you were expecting it to.

(but you're right, this is hacky)